massmigration/migrations.py

```diff
@@ -59,53 +90,62 @@ def get_backend(self):
         backend_class = import_string(self.backend_str)
         return backend_class()
 
-    def launch(self):
+    def launch(self, db_alias=None):
         """ Pass the migration to the backend to perform the data operation(s).
             This is what should be called by the web interface to trigger the migration.
         """
-        self.check_dependencies()
+        allowed_db_aliases = self.get_allowed_db_aliases()
+        if db_alias not in allowed_db_aliases:
+            raise CannotRunOnDB(
+                f"Migration {self.key} can't run on {self.database_alias}. "
+                f"The available dbs are {', '.join(allowed_db_aliases)}. "
```

Comment:
```suggestion
                f"The allowed DBs for this migration are {', '.join(allowed_db_aliases)}. "
```

Just tweaking the wording to make it clear that we're talking about the allowed DBs for this migration, rather than all of the "available" ones.